servicetalk-concurrent-api/src/test/java/io/servicetalk/concurrent/api/TestCompletableTest.java

```diff
@@ -58,30 +61,68 @@ void testNonResubscribeableCompletable() {
     @Test
     void testSequentialSubscribeCompletable() {
         TestCompletable source = new TestCompletable.Builder()
+                .sequentialSubscribers()
                 .build();
 
         source.subscribe(subscriber1);
+        subscriber1.awaitSubscription();
         source.onComplete();
         subscriber1.awaitOnComplete();
 
         source.subscribe(subscriber2);
+        subscriber2.awaitSubscription();
         assertThat(subscriber2.pollTerminal(10, MILLISECONDS), is(nullValue()));
         source.onComplete();
         subscriber2.awaitOnComplete();
     }
 
+    @Test
+    void testSequentialSubscribePublisherCancelResetsSubscriber() {
+        SequentialCompletableSubscriberFunction sequentialCompletableSubscriberFunction =
+                new SequentialCompletableSubscriberFunction();
+        TestCompletable source = new TestCompletable.Builder()
+                .sequentialSubscribers(sequentialCompletableSubscriberFunction)
+                .build();
+
+        assertThat(sequentialCompletableSubscriberFunction.subscriber(), is(nullValue()));
+        assertThat(sequentialCompletableSubscriberFunction.isSubscribed(), is(false));
+
+        source.subscribe(subscriber1);
+        Cancellable cancellable1 = subscriber1.awaitSubscription();
+        assertThat(sequentialCompletableSubscriberFunction.subscriber(),
+                hasToString(containsString(subscriber1.toString())));
+        assertThat(sequentialCompletableSubscriberFunction.isSubscribed(), is(true));
+        cancellable1.cancel();
+        assertThat(sequentialCompletableSubscriberFunction.subscriber(), is(nullValue()));
+        assertThat(sequentialCompletableSubscriberFunction.isSubscribed(), is(false));
+        subscriber1.pollTerminal(10, MILLISECONDS);
+
+        source.subscribe(subscriber2);
+        subscriber2.awaitSubscription();
+        assertThat(sequentialCompletableSubscriberFunction.subscriber(),
+                hasToString(containsString(subscriber2.toString())));
+        assertThat(sequentialCompletableSubscriberFunction.isSubscribed(), is(true));
+        assertThat(subscriber2.pollTerminal(10, MILLISECONDS), is(nullValue()));
+        source.onComplete();
+        assertThat(sequentialCompletableSubscriberFunction.subscriber(), is(nullValue()));
+        assertThat(sequentialCompletableSubscriberFunction.isSubscribed(), is(false));
+        subscriber2.awaitOnComplete();
+    }
+
     @Test
     void testConcurrentSubscribeCompletable() {
         TestCompletable source = new TestCompletable.Builder()
                 .concurrentSubscribers()
```

Comment:
Note that `concurrentSubscribers()` mode isn't used anywhere outside these tests for `TestPublisher|Completable|Single`. Disabling auto on-subscribe won't have any impact.
Furthermore, we do not advertise `testFixtures` as stable public API.